### ISSUE TITLE
mediastream vs datachannels in "initial offers"

### ISSUE THREAD

**Issue by dontcallmedom** [rtcweb-wg/jsep#185]
The "Initial Offers" section does not distinguish systematically between requirements that apply for any type of offer and the ones that apply only to mediastreams.

For instance:

> The next step is to generate m= sections, as specified in [RFC4566]  Section 5.14, for each MediaStreamTrack that has been added to the   PeerConnection via the addStream method

I think this should apply both to mediastreamtrack and to datachannels. 

Then many of the rules described under "Each m= section MUST include the following attribute lines:" seem specific to mediastreams, but not all; and that set of rules is then followed by a statement that a m= section must be created for data channels.

This could use some reorganization and clarification.

**Comment by juberti** [comment]
Agree this could be refactored a bit. Magnus had a similar comment in the past.

**Comment by fluffy** [comment]
Might be nice to have a term that referred to a MST or DataChannel across all the specs

**Comment by hardie** [comment]
Suggest contact with Magnus or Jessep for potential language.

**Comment by juberti** [comment]
I think this is mostly editorial and as such probably needs to be resolved within the author team.

**Comment by taylor-b** [comment]
Can this be considered fixed now? The createOffer section has since been structured into "here's what you generate for each RtpTransceiver", "here's what you generate if there's a data channel". And the data channel requirements look complete to me:

>    Lastly, if a data channel has been created, a m= section MUST be
>    generated for data.  The <media> field MUST be set to "application"
>    and the <proto> field MUST be set to "UDP/DTLS/SCTP" if the default
>    candidate uses UDP transport, or "TCP/DTLS/SCTP" if the default
>    candidate uses TCP transport [I-D.ietf-mmusic-sctp-sdp].  The "fmt"
>    value MUST be set to "webrtc-datachannel" as specified in
>    [I-D.ietf-mmusic-sctp-sdp], Section 4.1.
> 
>    Within the data m= section, the "a=mid", "a=ice-ufrag", "a=ice-pwd",
>    "a=fingerprint", "a=dtls-id", and "a=setup" lines MUST be included as
>    mentioned above, along with an "a=fmtp:webrtc-datachannel" line and
>    an "a=sctp-port" line referencing the SCTP port number as defined in
>    [I-D.ietf-mmusic-sctp-sdp], Section 4.1.

**Comment by ekr** [comment]
I agree with @taylor-b. @juberti, shall we close this?

**Comment by juberti** [comment]
I think the concern was that the data channel paragraphs feel a bit like an afterthought. I think this could be improved, but I don't think the current situation is dire.

**Comment by fluffy** [comment]
I can live with the current text as is. Done is a feature and I don't think this will cause any interoperability problems.

**Comment by juberti** [comment]
This text has been improved as of PR #518. If we can resolve #528, and this text still reads well, I think we can close this issue.

**Comment by juberti** [comment]
Pending resolution of #528

**Comment by fluffy** [comment]
#528 is it's own bug. I think we an just close this and any changes needed can be dealt with by #528

**Comment by juberti** [comment]
I want to take one last editorial pass over the text once #528 lands. This is a purely editorial issue at this point.
